capstone-app/src/app/modules/history/wiki-media/wiki-media.component.ts

```diff
@@ -34,19 +47,69 @@ export class WikiMediaComponent implements OnChanges, OnInit {
   }
 
   getResults(queryString: string) {
-    this.wikiService.search(queryString)
+    this.wikiService.search(queryString, 'en', 'History')
       .subscribe((result: WikiServiceResult) => {
         if (result.history) {
           this.body = result.history;
           this.history = result.history;
           this.urls = result.furtherReading;
           this.title = result.title;
           this.loading = false;
+          this.langlinks = result.langlinks;
+          const language = this.langlinks.get('ca');
+          this.prevQuery = language.get('searchQuery');
+          this.prevWordForHistory = 'History';
+          this.prevPreFix = 'en';
         }
         else {
           this.error = 'API did not return a valid response.';
           console.error('API did not return a valid response.');
         }
     });
   }
+
+  changeLanguage(queryString: string, languagePrefix: string, wordForHistory: string) {
+    this.wikiService.searchNewLang(queryString, languagePrefix, wordForHistory)
+      .subscribe((result: WikiServiceResult) => {
+        this.history = result.history;
+        if (!(this.history) || (this.history.includes("CPU time usage:"))){
+          this.loading = true;
+          this.error = "Unfortunately, the language you requested does not have an available tranlation for this page. Please select a different language.";
```

Comment:
I'm a little worried about this object. I'd assume that for many of the languages listed, there may not be a "history" section, but there still would be informaiton, such as https://mg.wikipedia.org/wiki/Los_Angeles. Wouldn't it make more sense to include the intro paragraphs if there's no history section found?

Reply:
Logic* not object

Reply:
Yes, this actually would still do that. The var name is a little confusing, but the history string is returned with either the history section, or if that is null, history is returned with the beginning paragraphs of the wiki page. The only reason history would be null is if there was no text response on the wiki pg, or it loaded but didn't have any data in which case it returns cpu stats.

Reply:
Can you go to the 'mg' translation of Los Angeles on your branch? The wikipedia article has information but it's displaying 'The Malagasy translation for Los Angeles is unavailable.
Unfortunately, the language you requested does not have an available tranlation for this page. Please select a different language' on our website

Reply:
Hmm, yes I'll look into that. It might be that sometimes there is data and a cpu statistic is returned?